lib/elixir_make/artefact.ex

```diff
@@ -187,22 +216,48 @@ defmodule ElixirMake.Artefact do
   @doc """
   Returns the url for the current target.
   """
-  def current_target_url(config, precompiler, nif_version) do
+  def current_target_url(config, precompiler, current_nif_version) do
     case precompiler.current_target() do
       {:ok, current_target} ->
-        available_urls = available_target_urls(config, precompiler)
-        target_at_nif_version = {current_target, nif_version}
+        nif_versions =
+          config[:make_precompiler_nif_versions] ||
+            [versions: []]
+
+        versions = nif_versions[:versions]
+        fallback_version = nif_versions[:fallback_version]
+
+        nif_version_to_use =
+          case find_nif_version(current_nif_version, versions) do
+            :no_candidates ->
+              if is_function(fallback_version, 3) do
+                fallback_version.(current_target, current_nif_version, versions)
+              else
+                current_nif_version
+              end
+
+            version when is_binary(version) ->
+              version
+          end
 
-        case List.keyfind(available_urls, target_at_nif_version, 0) do
-          {^target_at_nif_version, download_url} ->
-            {:ok, current_target, download_url}
+        case nif_version_to_use do
+          :compile ->
+            {:error, :compile_from_source}
+
```

Comment:
Do you think we need this clause? If we don't have this clause, it would return `{:error, {:unavailable_target, ...}}`, which would translate to a compile anyway.

Reply:
I was thinking about this too. Adding this was to suppress the warning message below, but I guess it's not really a problem if we show this in the log?

Reply:
I think the error message is correct, that we indeed did not find something precompiled, and that you need to compile from source. I think letting the user know there is no precompiled is useful feedback? If someone complains add the feature later :D

Reply:
No problem~ I'll change this back!